### ISSUE TITLE
Memory issues in matrix_multiply example

### ISSUE THREAD

**Issue by gunan** [codeplaysoftware/computecpp-sdk#7]
Reproduced this on both Hawaii and Fiji GPUs. When given matrix size is not a power of two, it fails.
Steps to reproduce:

```
$ git clone https://github.com/codeplaysoftware/computecpp-sdk
$ cd computecpp-sdk/samples
$ cmake . -DCOMPUTECPP_PACKAGE_ROOT_DIR=/usr/local
$ cd matrix_multiply
$ ./matrix_multiply 32
 Input matrix 
C++: Time: 0
GFLOPs: inf
 The Device Max Work Group Size is : 256
 The order is : 32
 The blockSize is : 8
SYCL: Time: 24
GFLOPs: 0.00273067
 Output 
Success
$ ./matrix_multiply 33
 Input matrix 
C++: Time: 1
GFLOPs: 0.071874
*** Error in `./matrix_multiply': free(): invalid next size (normal): 0x00000000015a7d40 ***
Aborted (core dumped)
```

I am hoping this is only a problem in the sample?

**Comment by DuncanMcBain** [comment]
Thanks for your report. We should have this fixed shortly.

**Comment by DuncanMcBain** [comment]
Thanks Gunan, this was fixed in #8.
